src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -176,6 +177,11 @@ def module_version(self):
     def product_version(self):
         return _get_core_version()
 
+    @property
+    def extension_version(self):
+        from azure.cli.core.extension import get_extension
+        return get_extension(self.extension_name).version if self.extension_name else None
```

Comment:
Move this logic to the place where `set_command_details` is called. The `telemetry` module should not have knowledge (dependencies) of extension's logic.